### PR TITLE
Dom/no implicit any

### PR DESCRIPTION
**Pull request by domoritz**
New version of #1649

### CODE REVIEW

File: src/scale.ts [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
+    export const YEAR: 'year' = 'year';
 }
 
-export enum BandSize {
-  FIT = 'fit' as any
+export type NiceTime = typeof NiceTime.SECOND | typeof NiceTime.MINUTE | typeof NiceTime.HOUR
+  | typeof NiceTime.DAY | typeof NiceTime.WEEK | typeof NiceTime.MONTH | typeof NiceTime.YEAR;
+
+export namespace BandSize {
+  // FIXME: should not be any
```
Comment (kanitw) [inline src/scale.ts]: Not sure why you add `any` and FIXME here?  Why not just make it right?

File: vega-lite-schema.json [hunk 2]
```diff
@@ -1655,15 +1651,7 @@
...
-                    "minimum": 0,
-                    "anyOf": [
-                        {
-                            "$ref": "#/definitions/BandSize"
-                        },
-                        {
-                            "type": "number"
-                        }
-                    ]
+                    "minimum": 0
```
Comment (kanitw) [inline vega-lite-schema.json]: Why does anyOf disappear here?   (Probably related to that FIT is `any` now?)

File: vega-lite-schema.json [hunk 3]
```diff
@@ -2020,7 +2008,7 @@
                 },
                 "titleFontWeight": {
                     "description": "The font weight of the legend title.",
-                    "type": "string"
+                    "type": "number"
```
Comment (kanitw) [inline vega-lite-schema.json]: I guess this is either number of string?  (500 or "bold" are both fine?)

File: vega-lite-schema.json [hunk 4]
```diff
@@ -560,11 +560,11 @@
                         false,
                         "year",
                         "month",
-                        "week",
                         "day",
                         "second",
                         "minute",
-                        "hour"
+                        "hour",
+                        "week"
```
Comment (domoritz) [inline vega-lite-schema.json]: Why is there still week?

**Comment by kanitw** [comment]
@weswigham  Your contribution just got merged!  Thank you :)

**Comment by weswigham** [comment]
@kanitw Thanks! I may look into enabling strict null checks in the future if I have time - it seems like you guys could really use making null/undefined explicit in the type signatures of your API if you're using it to signal things!

**Comment by kanitw** [comment]
Sure it will be helpful to do so but we just haven't done it because it takes time and not our top priority yet.

**Pull request merged**